GhostNetwork.Profiles.Api/Controllers/FriendController.cs

```diff
@@ -0,0 +1,142 @@
+            [FromQuery, Range(0, int.MaxValue)] int skip,
+            [FromQuery, Range(1, 100)] int take,
+            [FromRoute] Guid userId)
+        {
+            var (friends, totalCount) = await friendService.SearchFriendsAsync(skip, take, userId);
+            Response.Headers.Add("X-TotalCount", totalCount.ToString());
+
+            return Ok(friends);
+        }
+
+        /// <summary>
+        /// Search received friend requests by userId
+        /// </summary>
+        /// <param name="skip">Skip friends up to a specified position</param>
+        /// <param name="take">Take friends up to a specified position</param>
+        /// <param name="userId">Filters by userId</param>
+        /// <response code="200">Return received friend requests</response>
+        [HttpGet("followers/{userId}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [SwaggerResponseHeader(StatusCodes.Status200OK, "X-TotalCount", "Number", "Total number of friend requests")]
+        public async Task<ActionResult<IEnumerable<Friend>>> SearchFriendRequestsAsync(
+            [FromQuery, Range(0, int.MaxValue)] int skip,
+            [FromQuery, Range(1, 100)] int take,
+            [FromRoute] Guid userId)
+        {
+            var (friends, totalCount) = await friendService.SearchFriendRequestsAsync(skip, take, userId);
+            Response.Headers.Add("X-TotalCount", totalCount.ToString());
+
+            return Ok(friends);
+        }
+
+        /// <summary>
+        /// Search sended friend requests by userId
+        /// </summary>
+        /// <param name="skip">Skip friends up to a specified position</param>
+        /// <param name="take">Take friends up to a specified position</param>
+        /// <param name="userId">Filters friends by userId</param>
+        /// <response code="200">Return sended friend requests</response>
+        [HttpGet("followed/{userId}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [SwaggerResponseHeader(StatusCodes.Status200OK, "X-TotalCount", "Number", "Total number of friend requests")]
+        public async Task<ActionResult<IEnumerable<Friend>>> SearchSendedFriendRequestsAsync(
+            [FromQuery, Range(0, int.MaxValue)] int skip,
+            [FromQuery, Range(1, 100)] int take,
+            [FromRoute] Guid userId)
+        {
+            var (friends, totalCount) = await friendService.SearchSentFriendRequestsAsync(skip, take, userId);
+            Response.Headers.Add("X-TotalCount", totalCount.ToString());
+
+            return Ok(friends);
+        }
+
+        /// <summary>
+        /// Send friend request
+        /// </summary>
+        /// <param name="fromUser">Filters friend request by fromUser</param>
+        /// <param name="toUser">Filters friend request by toUser</param>
+        /// <response code="200">Friend request successfully created</response>
+        [HttpPost("friendrequest/{fromUser}/{toUser}")]
```

Comment:
Url looks strange /friends/friendrequest/{fromUser}/{toUser}
Maybe it is better to have something like /friends/{fromUser}/{toUser}?